includes/functions.php

```diff
@@ -2080,86 +2074,88 @@ function tsml_import_has_changes($data_source_url, $meetings, $data_source_count
+			$meeting_update_date = date("M j, Y  g:i a", $current_meeting_last_update);
 			
 			if ($is_matched) {
-				$message_lines[] = "<tr style='color:gray;'><td>Change</td><td >$mtg_name</td><td>$dayofweek</td><td>$mtg_updte</td></tr>";
+				$message_lines[] = "<tr style='color:gray;'><td>Change</td><td >$meeting_name</td><td>$dayofweek</td><td>$meeting_update_date</td></tr>";
 			}
 			else {
-				$message_lines[] = "<tr style='color:green;'><td>Add New</td><td >$mtg_name</td><td>$dayofweek</td><td>$mtg_updte</td></tr>";
+				$message_lines[] = "<tr style='color:green;'><td>Add New</td><td >$meeting_name</td><td>$dayofweek</td><td>$meeting_update_date</td></tr>";
 			}
 		}
 	}
-
-	sort($feed_slugs);
-
-	// list removed meetings found in local db
+	
+	// list meetings in local database which are not matched with feed
 	foreach ($db_meetings as $db_meeting) {
 
-	    $mtg_slug =  $db_meeting['slug'];
-		if (is_array($db_meeting['day'])) {
-			$dow = implode("",  $db_meeting['day']);
-		} else {
-			$dow = $db_meeting['day'];
-		}
+		$dayofweek = $dow_number = '';
+		tsml_get_day_of_week_info( $db_meeting['day'], $dayofweek, $dow_number);
+	    $meeting_slug = $db_meeting['slug'];
 
-	    $is_matched = in_array( $mtg_slug, $feed_slugs );
+	    $is_matched = in_array( $meeting_slug, $feed_slugs );
 
 		if (!$is_matched) {
-		    $is_there_change = true;
-			if ( is_numeric( $dow )) {
-				$dayofweek = $days[date($dow)]; 
-			} else {
-				$dayofweek = $dow; 
-			}
-			$updated = $data_source_last_refresh;
-			$cur_meeting_lastupdate = strtotime($updated);
-			$mtg_updte = date("M j, Y  g:i a", $updated);
-			$mtg_name = $db_meeting['name'];
-			$message_lines[] = "<tr style='color:red;'><td>Remove</td><td >$mtg_name</td><td>$dayofweek</td><td>* $mtg_updte</td></tr>";
+			$meeting_update_date = date("M j, Y  g:i a", $data_source_last_refresh);
+			$meeting_name = $db_meeting['name'];
+			$message_lines[] = "<tr style='color:red;'><td>Remove</td><td >$meeting_name</td><td>$dayofweek</td><td>* $meeting_update_date</td></tr>";
 		}
 	}
+	
+	return $message_lines;
+}
+
+//function:	Returns corresponding day of week string and number for the day input
+if (!function_exists('tsml_get_day_of_week_info')) {
+	function tsml_get_day_of_week_info( $meeting_day_input, &$dayofweek, &$dow_number )
+	{
+		global $tsml_days;
+		
+		// when day is like "Monday" convert to number
+		if ( in_array($meeting_day_input, $tsml_days) ) {
+			$dow_number = date('N', strtotime( $meeting_day_input ) );
+			if ($meeting_day_input == 'Sunday') {$dow_number = '0';}
 
-	if ($is_there_change) {
-		$meetings_updated = $message_lines;
+		} elseif ( is_array($meeting_day_input) ) {
+			$dow_number = implode("",  $meeting_day_input);
+
+		} else {
+			$dow_number = $meeting_day_input;
+		}
+
+		// only accept valid day of week numbers
+		if (($dow_number >= 0) && ($dow_number <= 6)) {
+		    $dayofweek = $tsml_days[$dow_number];
+		} else {
+		    $dayofweek = 'invalid value';
+		}
```

Comment:
would be cleaner to say
```
if (array_key_exists($dow_number, $tsml_days)) {
```
or the whole thing could be
```
$dayofweek = array_key_exists($dow_number, $tsml_days) ? $tsml_days[$dow_number] : 'invalid value';
```

Reply:
also if you're going with the `list($dayofweek, $dow_number)` suggestion above, here you'd put a

`return [$dayofweek, $dow_number];`

Reply:
Yea, I'm definitely old school here. Ternary operators don't seem clearer to me, but if that's what you all prefer I'll work at it a bit more.
I'll take another run at trying to get ### list($dayofweek, $dow_number) / return [$dayofweek, $dow_number]; working again. Your coding example is pretty much exactly what I tried first and had trouble with. I did testing all afternoon with the 2nd method that I used, so I have some confidence in it.

Reply:
Seems my list problem centered around how I was initializing the variables. Working well now...